specs/PeriodicBackgoundSyncAndBackgroundSync.md

```diff
@@ -0,0 +1,581 @@
+  HRESULT Invoke([in] HRESULT errorCode, [in] BOOL result);
+}
+```
+
+C#
+```c#
+namespace Microsoft.Web.WebView2.Core
+{
+    enum CoreWebView2ServiceWorkerSynchronizationKind
+    {
+        // Indicates that the synchronization is a background synchronization.
+        // See [Background Synchronization](https://developer.mozilla.org/docs/Web/API/Background_Synchronization_API)
+        // for more information.
+        BackgroundSync = 0,
+        // Indicates that the synchronization is a periodic background synchronization.
+        // See [Periodic Background Synchronization](https://developer.mozilla.org/docs/Web/API/Web_Periodic_Background_Synchronization_API)
+        // for more information.
+        PeriodicSync = 1,
+    };
+
+    runtimeclass CoreWebView2ServiceWorkerRegistration
+    {
+        CoreWebView2ServiceWorkerSyncRegistrationManager SyncRegistrationManager { get; };
+    }
+
+    runtimeclass CoreWebView2ServiceWorker
+    {
+        Windows.Foundation.IAsyncOperation<Boolean> DispatchPeriodicSyncEventAsync(String Tag);
+    }
+
+    runtimeclass CoreWebView2ServiceWorkerSyncRegistrationManager
+    {
+        
+        event Windows.Foundation.TypedEventHandler<CoreWebView2ServiceWorkerSyncRegistrationManager, CoreWebView2ServiceWorkerSyncRegisteredEventArgs> BackgroundSyncRegistered;
+
+        event Windows.Foundation.TypedEventHandler<CoreWebView2ServiceWorkerSyncRegistrationManager, CoreWebView2ServiceWorkerSyncRegisteredEventArgs> PeriodicSyncRegistered;
+
+        Windows.Foundation.IAsyncOperation<IVectorView<CoreWebView2ServiceWorkerSyncRegistrationInfo>> GetSyncRegistrationsAsync(CoreWebView2ServiceWorkerSynchronizationKind Kind);
+    }
+
+    runtimeclass CoreWebView2ServiceWorkerSyncRegisteredEventArgs
+    {
+        CoreWebView2ServiceWorkerSyncRegistrationInfo RegistrationInfo { get; };
+    }
+
+    runtimeclass CoreWebView2ServiceWorkerSyncRegistrationInfo
+    {
+        Int64 MinInterval { get; };
+        String Tag { get; };
+    }
+}
+```
+
+
+# Examples
+### C++ Sample
+```cpp
+static constexpr WCHAR c_samplePath[] = L"ScenarioServiceWorkerSyncRegistrationManager.html";
+
+ScenarioServiceWorkerSyncRegistrationManager::ScenarioServiceWorkerSyncRegistrationManager(
+    AppWindow* appWindow)
+    : m_appWindow(appWindow), m_webView(appWindow->GetWebView())
+{
+    m_sampleUri = m_appWindow->GetLocalUri(c_samplePath);
+
+    auto webView2_13 = m_webView.try_query<ICoreWebView2_13>();
+    CHECK_FEATURE_RETURN_EMPTY(webView2_13);
+
+    wil::com_ptr<ICoreWebView2Profile> webView2Profile;
+    CHECK_FAILURE(webView2_13->get_Profile(&webView2Profile));
+    auto webViewProfile3 = webView2Profile.try_query<ICoreWebView2Profile3>();
+    CHECK_FEATURE_RETURN_EMPTY(webViewProfile3);
+    wil::com_ptr<ICoreWebView2ServiceWorkerManager> serviceWorkerManager;
+    CHECK_FAILURE(webViewProfile3->get_ServiceWorkerManager(&serviceWorkerManager));
+
+    if (serviceWorkerManager)

```

Comment:
You call CHECK_FAILURE, so can we remove the if (serviceWorkerManager) check? Is it valid for get_ServiceWorkerManager to return S_OK and provide a nullptr serviceWorkerManager? If not then just rely on CHECK_FAILURE and can remove if check.